clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareFetchRequestManager.java

```diff
@@ -34,30 +54,221 @@ public class ShareFetchRequestManager implements RequestManager {
+        this.metricsManager = metricsManager;
+        this.sessionHandlers = new HashMap<>();
+        this.nodesWithPendingFetchRequests = new HashSet<>();
     }
 
     @Override
     public NetworkClientDelegate.PollResult poll(long currentTimeMs) {
-        return EMPTY;
+        List<NetworkClientDelegate.UnsentRequest> requests;
+        Map<Node, ShareSessionHandler.ShareFetchRequestData> shareFetchRequests = prepareShareFetchRequests();
+
+        requests = shareFetchRequests.entrySet().stream().map(entry -> {
+            final Node target = entry.getKey();
+            final ShareSessionHandler.ShareFetchRequestData data = entry.getValue();
+            final ShareFetchRequest.Builder request = createShareFetchRequest(target, data);
+            final BiConsumer<ClientResponse, Throwable> responseHandler = (clientResponse, error) -> {
+                if (error != null) {
+                    handleShareFetchFailure(target, data, error);
+                } else {
+                    handleShareFetchSuccess(target, data, clientResponse);
+                }
+            };
+            return new NetworkClientDelegate.UnsentRequest(request, Optional.of(target)).whenComplete(responseHandler);
+        }).collect(Collectors.toList());
+
+        return new NetworkClientDelegate.PollResult(requests);
+    }
+
+    private void handleShareFetchSuccess(Node fetchTarget, ShareSessionHandler.ShareFetchRequestData data, ClientResponse resp) {
+        try {
+            final ShareFetchResponse response = (ShareFetchResponse) resp.responseBody();
+            final ShareSessionHandler handler = sessionHandler(fetchTarget.id());
+
+            if (handler == null) {
+                log.error("Unable to find ShareSessionHandler for node {}. Ignoring fetch response.",
+                        fetchTarget.id());
+                return;
+            }
+
+            final short requestVersion = resp.requestHeader().apiVersion();
+
+            if (!handler.handleResponse(response, requestVersion)) {
+                if (response.error() == Errors.FETCH_SESSION_TOPIC_ID_ERROR) {
+                    metadata.requestUpdate(false);
+                }
+
+                return;
+            }
+
+            final Map<TopicIdPartition, ShareFetchResponseData.PartitionData> responseData = new LinkedHashMap<>();
+            Map<Uuid, String> topicNames = handler.sessionTopicNames();
+
+            response.data().responses().forEach(topicResponse -> {
+                String name = topicNames.get(topicResponse.topicId());
+                if (name != null) {
+                    topicResponse.partitions().forEach(partition ->
+                            responseData.put(new TopicIdPartition(topicResponse.topicId(), partition.partitionIndex(), name), partition));
+                }
+            });
+            final Set<TopicPartition> partitions = responseData.keySet().stream().map(TopicIdPartition::topicPartition).collect(Collectors.toSet());
```

Comment:
This part looks temporary too. I think we are only building the set of TopicPartition because the FetchMetricsAggregator needs it.